tests/test_timer_integration.py

```diff
@@ -0,0 +1,31 @@
+"""Integration tests for the timer."""
+
+
+import asyncio
+from datetime import timedelta
+
+import async_solipsism
+import pytest
+
+from frequenz.channels.timer import Timer
+
+
+@pytest.mark.integration
+async def test_timer_timeout_reset(
+    event_loop: async_solipsism.EventLoop,  # pylint: disable=redefined-outer-name
+) -> None:
+    """Test that the receiving is properly adjusted after a reset."""
+
+    async def timer_wait(timer: Timer) -> None:
+        await timer.receive()
+
+    async with asyncio.timeout(2.0):
+        async with asyncio.TaskGroup() as task_group:
+            timer = Timer.timeout(timedelta(seconds=1.0))
+            start_time = event_loop.time()
+            task_group.create_task(timer_wait(timer))
+            await asyncio.sleep(0.5)
+            timer.reset()
+
+    run_time = event_loop.time() - start_time
+    assert run_time >= 1.5
```

Comment:
I'd also check for run_time < 2.0
```suggestion
    assert 1.5 >= run_time < 2.0
```

Reply:
Well, the whole block is inside a `with asyncio.timeout(2)`, so it will never be `> 2.0`, as in that case the `async with` block it will raise an exception.

Reply:
oh right, my comment does not apply